packages/app-project/src/shared/components/CollectionsModal/components/SelectCollection/SelectCollection.spec.js

```diff
@@ -1,94 +1,38 @@
 
-import SelectCollection from './SelectCollection'
-
-let wrapper
+import { mockCollections } from '../../CollectionsModal.stories.js'
+import SelectCollection from './SelectCollection.js'
 
 describe('Component > SelectCollection', function () {
-  const collections = [
-    { id: '1', display_name: 'Test One' },
-    { id: '2', display_name: 'Test Two' }
-  ]
   const onSearch = sinon.stub()
-  const onSubmit = sinon.stub()
 
-  before(function () {
-    wrapper = shallow(
+  beforeEach(function () {
+    render(
       <SelectCollection
-        collections={collections}
+        collections={mockCollections}
         onSearch={onSearch}
-        onSubmit={onSubmit}
+        userID='123'
       />
     )
   })
 
-  it('should render without crashing', function () {
-    expect(wrapper).to.be.ok()
-  })
-
-  it('should call the onSubmit callback on submit', function () {
-    wrapper.find('Grid').simulate('submit')
-    expect(onSubmit).to.have.been.calledOnce()
+  it('should be empty by default', function () {
+    const selectInput = document.querySelector('input')
+    expect(selectInput.value).to.equal('')
   })
 
-  describe('collections search input', function () {
-    let select
-
-    before(function () {
-      select = wrapper.find(Select)
-    })
-
-    it('should exist', function () {
-      expect(select).to.have.lengthOf(1)
-    })
-
-    it('should be empty by default', function () {
-      expect(select.prop('value')).to.eql({})
-    })
-
-    it('should list collections', function () {
-      expect(select.prop('options')).to.eql(collections)
-    })
-
-    it('should call the onSearch callback', function () {
-      const searchText = 'Hello'
-      const searchSpy = onSearch.withArgs({
-        favorite: false,
-        current_user_roles: 'owner,collaborator,contributor',
-        search: searchText
-      })
-      select.simulate('search', searchText)
-      expect(searchSpy).to.have.been.calledOnce()
-    })
-    it('should display the selected collection', function () {
-      const collection = { id: '1', display_name: 'Selected collection' }
-      wrapper.setProps({ selected: collection })
-      select = wrapper.find(Select)
-      expect(select.prop('value')).to.eql(collection)
-    })
+  it('should contain an Add button', function () {
+    expect(screen.getByText('CollectionsModal.SelectCollection.addButton')).exists()
   })
 
-  describe('Add button', function () {
-    let button
-
-    before(function () {
-      button = wrapper.find(Button)
-    })
-
-    it('should contain a button', function () {
-      expect(button).to.have.lengthOf(1)
-    })
+  it('should call the onSearch callback', async function () {
+    const user = userEvent.setup({ delay: null })
 
-    it('should submit the form', function () {
-      expect(button.props().type).to.equal('submit')
-    })
+    const selectInput = document.querySelector('input')
+    await user.type(selectInput, 'coll')
```

Comment:
@eatyourgreens this PR is almost ready for final review. I wanted to convert the relevant SelectCollection unit tests to RTL, but I'm having trouble with this line. In this unit test I want to type a string into the < input > element, but always get two errors: `[TypeError: activeElement.attachEvent is not a function]` and `[TypeError: activeElement.detachEvent is not a function]`

Are you familiar with [user.type](https://testing-library.com/docs/ecosystem-user-event/#typeelement-text-options) and why it might be throwing errors?

Reply:
I'm not. What is `activeElement` set to when that error throws?

Reply:
Reading quickly through the code, I don't see the tests waiting for the drop menu to open, so that's probably the cause of the error.

Reply:
`attachEvent` was a Microsoft API, in IE, so that’s weird too. The standard DOM element API is [addEventListener](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener).

Reply:
The text input that calls the `onSearch` callback is this:

https://github.com/grommet/grommet/blob/fb103df0846bbda1f7cb75ecf678d2f22daedc5d/src/js/components/Select/SelectContainer.js#L364-L381

That's rendered inside a `Drop` component, so it won't be in the DOM when your test first renders.

Reply:
That test relies on internal implementation details for the text combobox, that I added to Grommet here:
https://github.com/grommet/grommet/pull/6079

Reply:
By the way, the menu options show the collection ID, not the owner name (eg. 'Collection: 10), but I assume this is still WIP.

Reply:
`10` in the display name is arbitrary and a result of creating mock collections in the .stories file. The owner id is `123`. I'll change the mock collections to prevent confusion.

Reply:
> Try this test...

This test does work thanks! However, it still contains the activeElement error. When I log the active element in the unit test, it is the Select component, and googling similar errors didn't yield any help. If the test still passes and correctly calls `onSearch`, do you think it's worth further investigation?

Reply:
If the test fails when you type three characters, then I think it's working.